timesketch/__init__.py

```diff
@@ -170,7 +171,13 @@ def load_user(user_id):
         return User.query.get(user_id)
 
     # Setup CSRF protection for the whole application
-    CSRFProtect(app)
+    csrf = CSRFProtect(app)
+
+    # Load the tsctl API endpoints only if the TSCTL API key is set
+    if u'TSCTL_API_KEY' in app.config and app.config[u'TSCTL_API_KEY']:
```

Comment:
I would rather implement a general API KEY feature instead of this special case. You can achieve the same with a service account and wrap the POSTs with scraping the CSRF token. I know this is not elegant, but it works before we have proper API KEYs implemented. Take a look at this example:

https://gist.github.com/berggren/365d4e6a74eb245f6f6a019c43a8103c